### PR TITLE
Fix issue where screenshare dialogs eats user input across the whole web page

### PR DESCRIPTION
**Pull request by JamesBurnside**
# What
Host the screen share dialog model in a div element inside the composite

# Why
For some users, the screenshare dialog was eating input across the whole screen.
This was because the body html element (that the modal dialog is hosted in) had:
```css
body { display: flex }
```
Which messed up Fluent's modal dialog hosting.

By instead hosting the dialog inside a component in our composite we get two benefits:
* We constrain the dialog to be within our composite
* We control the dialog's host div's css and hence customers won't hit this issue

# How Tested
In a test app. Was able to repro the problem then build a local package with these changes and test that the screenshare dialog no longer eats all input.

### CODE REVIEW

File: packages/react-composites/src/composites/CallComposite/CallScreen.tsx [hunk 1]
```diff
@@ -115,6 +115,7 @@ export const CallScreen = (props: CallScreenProps): JSX.Element => {
     }
   }
 
+  const screenShareHostHostId = 'UILibraryMediaGallery';
```
Comment (alcail) [inline packages/react-composites/src/composites/CallComposite/CallScreen.tsx:118]: nit: why the double 'Host' in the name?

**Review by alcail — APPROVED** [review]

File: packages/react-composites/src/composites/CallComposite/CallScreen.tsx (hunk 1 above)
Comment (JamesBurnside) [reply packages/react-composites/src/composites/CallComposite/CallScreen.tsx:118]: typo! :D I'll update that

Comment (JamesBurnside) [reply packages/react-composites/src/composites/CallComposite/CallScreen.tsx]: Updated!

File: packages/react-components/src/components/VideoGallery.tsx [hunk 2]
```diff
@@ -201,7 +201,7 @@ export const VideoGallery = (props: VideoGalleryProps): JSX.Element => {
   ]);
 
   if (shouldFloatLocalVideo()) {
-    const floatingTileHostId = 'UILibaryFloatingTileHost';
+    const floatingTileHostId = 'UILibraryFloatingTileHost';
```
Comment (anjulgarg) [inline packages/react-components/src/components/VideoGallery.tsx:204]: what are these ids for?

Comment (JamesBurnside) [reply packages/react-components/src/components/VideoGallery.tsx:204]: They are referenced by the layerprop hostid: https://developer.microsoft.com/en-us/fluentui#/controls/web/layer#ILayerProps

For fluent modal dialogs they essentially say, render the element in the element with this ID and not in the default location (which is at the end of the DOM directly under body)

**Review by anjulgarg — APPROVED** [review]

**Pull request merged**